config/checker-framework-suppressions/checker-index-suppressions.xml

```diff
@@ -1,4 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
+
+<!DOCTYPE suppressions PUBLIC
+        "-//Checkstyle//DTD SuppressionFilter Configuration 1.1//EN"
+        "https://checkstyle.org/dtds/suppressions_1_1.dtd">
```

Comment:
This is not a checkstyle suppression file.

Reply:
Should I create a new file and add these suppressions in that case??
Or please suggest if needed to proceed in any other way?....

Reply:
You have to update this file, but correctly. It is not valid XML to put a doctype in the middle of the XML. This is not a checkstyle suppression file, so this kind of content is incorrect here.

CI generates the patches file which you can copy and download and apply to your local to pass. However, since this current CI run is invalid, I don't think it generated the correct patch. You can use the old links I pasted a few posts above instead.

Reply:
The links that you have posted give empty patch files.
However, I have downloaded multiple of those from other merged PRs but none of them seem to apply in my local....

![image](https://github.com/checkstyle/checkstyle/assets/148562451/a199b360-88d5-45d3-8789-cd1f23e4db88)

Reply:
Since the error is 'DOCTYPE is disallowed when the feature "http://apache.org/xml/features/disallow-doctype-decl" set to true' , won't **_changing the setFeature to false_** work ?.. 
please do correct me in case my assumption is wrong.

Reply:
First, look at the XML files you created. They are not valid as they have 2 top level tags. XML files can only have 1 top level tag.
Second, look at the DTD you added. You added the checkstyle's main suppression DTD, https://checkstyle.org/filters/suppressionfilter.html#SuppressionFilter. Once again, this XML is not that kind of suppression file. The suppression file used here is a custom creation only for checker.
Third look of the format of the tags and such before you made any changes to this file. You will notice the main tag is `checkerFrameworkError` with 5 children. None of them are `suppress`.

Again, this is McDonalds and you are trying to order Burger King.

I am not sure why you patches are not applying, but I can say they are not empty.
https://github.com/checkstyle/checkstyle/actions/runs/7648358374?pr=14347
> checker-index-patch	8.89 KB 	
checker-methods-resource-fenum-patch	2.06 KB 

Your patch is probably failing as you are trying to apply them on top of the changes you made the suppression file already. Revert all the changes to the suppression file you made, and then try to apply them.

There is not much more I can do if you still cannot apply the patch or proceed to make CI green.